frontend/projects/upgrade/src/app/features/dashboard/experiments/modals/upsert-decision-point-modal/upsert-decision-point-modal.component.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  UPSERT_EXPERIMENT_ACTION,
+  UpsertDecisionPointParams,
+  DecisionPointFormData,
+  IContextMetaData,
+  ExperimentDecisionPoint,
+  Experiment,
+} from '../../../../../core/experiments/store/experiments.model';
+
+@Component({
+  selector: 'upsert-decision-point-modal',
+  imports: [
+    CommonModalComponent,
+    MatFormFieldModule,
+    MatInputModule,
+    MatCheckboxModule,
+    MatAutocompleteModule,
+    CommonModule,
+    ReactiveFormsModule,
+    TranslateModule,
+  ],
+  templateUrl: './upsert-decision-point-modal.component.html',
+  styleUrl: './upsert-decision-point-modal.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UpsertDecisionPointModalComponent implements OnInit, OnDestroy {
+  isLoadingUpsertDecisionPoint$ = this.experimentService.isLoadingExperiment$;
+  contextMetaData$ = this.experimentService.contextMetaData$;
+
+  subscriptions = new Subscription();
+  isPrimaryButtonDisabled$: Observable<boolean>;
+  isInitialFormValueChanged$: Observable<boolean>;
+
+  initialFormValues$ = new BehaviorSubject<DecisionPointFormData>(null);
+
+  // Filtered options for autocomplete
+  filteredSites$: Observable<string[]>;
+  filteredTargets$: Observable<string[]>;
+
+  decisionPointForm: FormGroup;
+  currentContext: string;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA)
+    public config: CommonModalConfig<UpsertDecisionPointParams>,
+    private formBuilder: FormBuilder,
+    private experimentService: ExperimentService,
+    private decisionPointHelperService: DecisionPointHelperService,
+    public dialogRef: MatDialogRef<UpsertDecisionPointModalComponent>
+  ) {}
+
+  ngOnInit(): void {
+    this.experimentService.fetchContextMetaData();
+    this.currentContext = this.config.params.context || '';
+    this.createDecisionPointForm();
+    this.setupAutocompleteFilters();
+
+    // Add listeners AFTER form is fully set up
+    this.listenForIsInitialFormValueChanged();
+    this.listenForPrimaryButtonDisabled();
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.unsubscribe();
+  }
+
+  createDecisionPointForm(): void {
+    const { sourceDecisionPoint, action } = this.config.params;
+    const initialValues = this.deriveInitialFormValues(sourceDecisionPoint, action);
+
+    this.decisionPointForm = this.formBuilder.group({
+      site: [initialValues.site, [Validators.required]],
+      target: [initialValues.target, [Validators.required]],
+      excludeIfReached: [initialValues.excludeIfReached],
+    });
+
+    this.initialFormValues$.next(this.decisionPointForm.value);
+  }
+
+  deriveInitialFormValues(
+    sourceDecisionPoint: ExperimentDecisionPoint,
+    action: UPSERT_EXPERIMENT_ACTION
+  ): DecisionPointFormData {
+    const site = action === UPSERT_EXPERIMENT_ACTION.EDIT ? sourceDecisionPoint?.site || '' : '';
+    const target = action === UPSERT_EXPERIMENT_ACTION.EDIT ? sourceDecisionPoint?.target || '' : '';
```

Comment:
```suggestion
    const site = action === UPSERT_EXPERIMENT_ACTION.EDIT && sourceDecisionPoint?.site ? sourceDecisionPoint.site : '';
    const target = action === UPSERT_EXPERIMENT_ACTION.EDIT && sourceDecisionPoint?.target ? sourceDecisionPoint.target : '';
```

this reads oddly to me with the double fallbacks to empty string... IMO if a ternary is going to be used, all logic should happen on the lefthand side so the conditionals can be simple values.